pkg/currency/service.go

```diff
@@ -17,13 +18,40 @@ const ExchangeRateDateFormat = time.RFC3339
 
 //go:generate mockery -name Service
 type Service interface {
+	Currencies() ([]string, error)
+	HasCurrency(currency string) (bool, error)
 	ToEur(float64, string) (float64, error)
 	ToUsd(float64, string) (float64, error)
 	ToCurrency(string, float64, string) (float64, error)
 }
 
 type CurrencyService struct {
-	redis redis.Client
+	redis      redis.Client
+	currencies []string
+	lck        sync.Mutex
+}
+
+func (service *CurrencyService) Currencies() ([]string, error) {
+	service.lck.Lock()
+	defer service.lck.Unlock()
+
+	if len(service.currencies) > 0 {
+		return service.currencies, nil
+	}
+
+	currencies, err := service.redis.HKeys(ExchangeRateDataKey)
+
+	if err != nil {
+		return nil, errors.WithMessage(err, "CurrencyService: error getting currency keys")
+	}
+
+	service.currencies = currencies
+
+	return service.currencies, nil
+}
+
+func (service *CurrencyService) HasCurrency(currency string) (bool, error) {
+	return service.redis.HExists(ExchangeRateDataKey, currency)
```

Comment:
moving the new functions beneath the new() function would be better I would assume